pkg/instancetype/apply/memory.go

```diff
@@ -74,3 +69,31 @@ func applyMemory(
 
 	return nil
 }
+
+func validateMemory(
+	baseConflict *conflict.Conflict,
+	instancetypeSpec *v1beta1.VirtualMachineInstancetypeSpec,
+	vmiSpec *virtv1.VirtualMachineInstanceSpec,
+) (conflicts conflict.Conflicts) {
+	if vmiSpec.Domain.Memory.Guest != nil {
```

Comment:
@lyarwood Why not make this check optional too? (see comment below, instancetypes can set this but don't have to?)

Reply:
@0xFelix I'm not sure I follow, this isn't make the check conditional it's asserting that any guest value raises a conflict that's the behaviour that we currently have. We've talked about making `instancetypeSpec.Memory.Guest` optional in a future version of the API but at the moment it's required and always set.

Reply:
Yeah IMHO moving everything to optional will require a new version.